File: CodingInterviewsPythonSolutions/Max Consecutive Ones III.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        sz = len(nums)
        result = 0
        start = 0
        zeroes = 0
        for end in range(sz):
            zeroes += nums[end] == 0
            while zeroes > k:
                zeroes -= nums[start] == 0
                start += 1
            result = max(result, end - start + 1)
        return result

File: CodingInterviewsJavaSolutions/Max Consecutive Ones III.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int sz = nums.length;
        int result = 0;
        int start = 0;
        int zeroes = 0;
        for (int end = 0; end < sz; end++) {
            zeroes += nums[end] == 0 ? 1 : 0;
            while (zeroes > k) {
                zeroes -= nums[start] == 0 ? 1 : 0;
                start++;
            }
            result = Math.max(result, end - start + 1);
        }
        return result;
    }
}
